pydantic/_internal/_decorators.py

```diff
@@ -786,3 +803,19 @@ def wrap_model_serializer_in_wrap_mode(
             raise TypeError(f'Unrecognized serializer signature for {serializer} with `mode={mode}`:{sig}')
         func = cast(AnyCoreSerializer, serializer)
         return func
+
+
+ModelType = TypeVar('ModelType', bound='BaseModel')
```

Comment:
Should we loosen the bound here? E.g., I'm thinking this might apply to a dataclass or whatever.

Reply:
Maybe, but then we end accepting `Callable[[Any, Any], Any]` in some situations lol. It's kind of a struggle to type these decorators, especially given how we let you include `info` or not and how we accept `classmethod[Any]` and such. I almost want to sit down with some of the typing folks at PyCon and see if they have any ideas. I do plan to do a bit more typing work on this PR, so I can play around with loosening the bound.

Reply:
I think we need to relax it to at least accept dataclass.

On this, I'd rather start loose and move to stricter types.

Reply:
relaxed